src/google/protobuf/compiler/objectivec/objectivec_message.h

```diff
@@ -66,6 +66,9 @@ class MessageGenerator {
   void GenerateExtensionRegistrationSource(io::Printer* printer);
   void DetermineForwardDeclarations(set<string>* fwd_decls);
 
+  // Checks if the message or a nested message includes a oneof definition.
+  bool IncludesOneOfDefinition() const;
```

Comment:
It is, it is used to pick off when to add the extra pragma ignore to the generated sources (direct ivar access)